src/mon/Monitor.cc

```diff
@@ -950,6 +950,9 @@ int Monitor::init()
   mgrmon()->prime_mgr_client();
 
   state = STATE_PROBING;
+
+  elector.notify_clear_peer_state();
```

Comment:
And I don't really get why we want to unconditionally clear peer state here. Doesn't...doesn't this throw away our connection scores even time the monitor boots? That's not what we want!

Reply:
My motivation for putting that there is for the case where we want to upgrade from a buggy version of Ceph where the Connection Scores are messed up to this current version, we don't have to rely on the user/admin to execute the command:

``ceph daemon connection scores reset``

I thought this was the best place to put it because the function is in the code path for when we upgrade the monitor. I know it overlaps with the scenario where the monitor is down or restart, but at the time I couldn't find a better place to put it. I don't think it will hurt to clear the scores and also reassign the rank of the peer_tracker to the actual rank according to Monmap every time the Monitor restarts.

In my opinion, I think the benefits here outweigh the cons when we throw away the scores and populate a new one since I think it is important for the patch to make sure we don't retain any issues from the previous buggy version when upgrading to this patch.

But ofc you have a better solution, I'm all ears.

Reply:
I'm not sure the right place to fix this, but as we said verbally, I don't think we can toss out the scores on every monitor boot. Besides being messy, I'm worried about bad behavior in degenerate cases.

Reply:
Yep will add a condition to only clear the peer_tracker if it is messed up beyond recovery.